src/app/pages/BridgeDepositPage/components/WalletSelector/index.tsx

```diff
@@ -35,11 +39,20 @@ export const WalletSelector: React.FC = () => {
   const bridgeLocked = checkMaintenance(States.BRIDGE);
 
   const dispatch = useDispatch();
-  const { ethereum } = window;
 
   const { bridgeChainId } = useSelector(selectWalletProvider);
+  const { wallet } = useContext(WalletContext);
   const { chain } = useSelector(selectBridgeDepositPage);
-  const chainId = parseInt(ethereum?.chainId as string);
+
+  const chainId = useMemo(() => {
```

Comment:
We have this exact code on 3 different places (and there is one place with a slightly different implementation). Could you please create a custom hook for that and use the hook?